### ISSUE TITLE
Variables object in plugin options?

### ISSUE THREAD

**Issue by dbooom** [pascalduez/postcss-map#15]
Is there any particular reason why `postcss-map` doesn't support variables object in options, like `postcss-css-variables` or `postcss-simple-vars` do?

**Comment by pascalduez** [comment]
Hi,

could you explain a bit more what you mean by "variables object in options" ?

**Comment by dbooom** [comment]
Hello.

Here's the links to plugins options I'm talking about:
https://github.com/MadLittleMods/postcss-css-variables#variables-default-
https://github.com/postcss/postcss-simple-vars#variables

So instead of declaring variables in YAML/JSON we have the ability to do it directly in JavaScript. It can be helpful for some dynamic data.

Thanks.

**Comment by pascalduez** [comment]
Right, I'll think of this a bit more. It could be an interesting addition.
But bear in mind `postcss-map` is not yet another variables module.

So roughly:
```js
var dummy = {
  dummy: {
    foo: 'foo',
    bar: 'bar'
  }
};

var opts = {
  basePath: 'css/',
  maps: [ dummy, 'breakpoints.yml', 'fonts.yml' ]
};
```

```css
.bar {
  content: map(dummy, bar);
}
```

**Comment by dbooom** [comment]
Looks great to me.

As an example use case (and why I asked in the first place) is color manipulation functions directly in JavaScript.

Let's say I have some basic color scheme declared in YAML/JSON. Currently it's impossible to add variations tied to these basic colors with libs like TinyColor etc. On the other hand with plain JavaScript we can create separate commonjs module, manipulate our color variables and then make them globally available through `postcss-map` plugin.

```javascript
var require colors = require('./colors');

var opts = {
  basePath: 'css/',
  maps: [ colors, 'breakpoints.yml', 'fonts.yml' ]
};
```

Cheers

**Comment by evanpaschke** [comment]
+1 This would be great. I'm processing a color theme config json through a color theme generator (using chroma.js to make highlights and shadows from a single color) and would like to use that generated map in css. Accepting a variable would have the benefit of not writing an intermediary processed color theme file for me.

**Comment by pascalduez** [comment]
I have a working implementation locally, I just need to find some time polishing/testing and then publish.

**Comment by pascalduez** [comment]
Available from `postcss-map` >= 0.6.0
